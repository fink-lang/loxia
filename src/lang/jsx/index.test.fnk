{fink2js} = import '../../testing'
{describe, it, expect, to_match_snapshot} = import '@fink/jest'


describe 'jsx', fn:
  it 'compiles', fn:
    expect
      fink2js '
        jsx1 = <a/>
        jsx2 = <b></b>
        jsx3 = <a foo bar="ni" />
        jsx4 = <a foo bar=\'ni\' />
        jsx5 = <a foo bar={1234} />
        jsx6 =
          <a>
            foo {ni}
            <b /> ham
            spam
            <c />
            ni
          </a>
        '
      to_match_snapshot



