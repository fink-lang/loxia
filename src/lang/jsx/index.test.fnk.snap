// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsx compiles 1`] = `
"const jsx1 = <a />;
const jsx2 = <b></b>;
const jsx3 = <a foo bar=\\"ni\\" />;
const jsx4 = <a foo bar=\\"ni\\" />;
const jsx5 = <a foo bar={1234} />;
const jsx6 = <a>
    foo {ni}
    <b /> ham
    spam
    <c />
    ni
  </a>;
Object.assign(module.exports, {
  jsx1,
  jsx2,
  jsx3,
  jsx4,
  jsx5,
  jsx6
});"
`;
